pkg/lib/serverTime.js

```diff
@@ -258,42 +266,43 @@
         return status;
     };
 
-    const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
-                                                { superuser: "try" });
+    function get_os_release() {
+        if (self.os_release)
+            return Promise.resolve(self.os_release);
+        else
+            return read_os_release().then(release => {
+                self.os_release = release;
+                return release;
+            });
+    }
+
+    function get_timesync_backend() {
+        if (self.backend)
+            return Promise.resolve(self.backend);
+        else
+            return get_os_release().then(release => {
+                const config = cockpit.manifests.base1.config;
+                let backend = "none";
+                if (release &&
+                    release.ID &&
+                    config &&
+                    config["timesync-backend"])
+                    backend = config["timesync-backend"][release.ID];
+                self.backend = backend;
+                return backend;
+            });
+    }
+
+    function get_custom_ntp_timesyncd() {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17992-20221213-115634-adc42487-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)